.gitignore

```diff
@@ -1,4 +1,14 @@
 dist/
 node_modules/
-.vscode/
-npm-debug.log
\ No newline at end of file
+npm-debug.log
+
+# Created by https://www.gitignore.io/api/visualstudiocode
+
+### VisualStudioCode ###
+.vscode/*
+!.vscode/settings.json
```

Comment:
settings.json is meant to represent per-project settings for vscode editors. They contain things like tabs-vs-whitespace. The rationale is that it makes sense to share them, because everyone should be using the same per-project settings files anwyay.